File: primes.py
import time
import math


def is_prime(n):
    # if n < 2:
    #     return False
    # if n == 2:
    #     return True
    # if n % 2 == 0:
    #     return False
    
    limit = int(math.sqrt(n)) + 1
    divisor = 3
    
    while divisor <= limit:
        if n % divisor == 0:
            return False
        divisor += 2
    
    return True


def main():
    start_time = time.time()
    
    time_limit = 10.0
    prime_counter = 0
    last_prime = 3
    
    n = 3
    while (time.time() - start_time) < time_limit:
        if is_prime(n):
            prime_counter += 1
            last_prime = n
        n += 2
    
    print(f"In {time_limit} seconds found {prime_counter} primes. Biggest is {last_prime}.")


if __name__ == "__main__":
    main()

File: src/bin/primes.rs
use std::time::Instant;


fn is_prime(n:&u64)->bool{
	let limit:u32 = (*n as f64).sqrt() as u32 +1;
	let mut divisor:u32 = 3u32;

	while divisor <= limit {
		if n % divisor == 0 {
			return false;
		}
		divisor += 2u32;
	}
	return true;
}

fn main(){
	let startTime = Instant::now();

	let timeLimit = 10f64;
	let mut primeCounter = 0u32;
	let mut lastPrime = 3u32;

	let mut n = 3u32;
	while startTime.elapsed().as_secs_f64() < timeLimit{
		if is_prime(&n){
			primeCounter += 1u32;
			lastPrime = n;
		}
		n += 2u32;
	}


	println!("In {timeLimit} seconds found {primeCounter} primes. Biggest is {lastPrime}.");
}